vault/resource_approle_auth_backend_role.go

```diff
@@ -0,0 +1,384 @@
+
+	if v, ok := d.GetOk("role_id"); ok {
+		log.Printf("[DEBUG] Writing AppRole auth backend role %q RoleID", path)
+		_, err := client.Logical().Write(path+"/role-id", map[string]interface{}{
+			"role_id": v.(string),
+		})
+		if err != nil {
+			return fmt.Errorf("Error writing AppRole auth backend role %q's RoleID: %s", path, err)
+		}
+		log.Printf("[DEBUG] Wrote AppRole auth backend role %q RoleID", path)
+	}
+
+	return approleAuthBackendRoleRead(d, meta)
+}
+
+func approleAuthBackendRoleRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	path := d.Id()
+
+	backend, err := approleAuthBackendRoleBackendFromPath(path)
+	if err != nil {
+		return fmt.Errorf("Invalid path %q for AppRole auth backend role: %s", path, err)
+	}
+
+	role, err := approleAuthBackendRoleNameFromPath(path)
+	if err != nil {
+		return fmt.Errorf("Invalid path %q for AppRole auth backend role: %s", path, err)
+	}
+
+	log.Printf("[DEBUG] Reading AppRole auth backend role %q", path)
+	resp, err := client.Logical().Read(path)
+	if err != nil {
+		return fmt.Errorf("Error reading AppRole auth backend role %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Read AppRole auth backend role %q", path)
+	if resp == nil {
+		log.Printf("[WARN] AppRole auth backend role %q not found, removing from state", path)
+		d.SetId("")
+		return nil
+	}
+	iPolicies := resp.Data["policies"].([]interface{})
+	policies := make([]string, 0, len(iPolicies))
+	for _, iPolicy := range iPolicies {
+		policies = append(policies, iPolicy.(string))
+	}
+
+	var cidrs []string
+	if resp.Data["bound_cidr_list"].(string) != "" {
+		cidrs = strings.Split(resp.Data["bound_cidr_list"].(string), ",")
+	}
+
+	secretIDTTL, err := resp.Data["secret_id_ttl"].(json.Number).Int64()
+	if err != nil {
+		return fmt.Errorf("Expected secret_id_ttl %q to be a number, isn't", resp.Data["secret_id_ttl"])
+	}
+
+	secretIDNumUses, err := resp.Data["secret_id_num_uses"].(json.Number).Int64()
+	if err != nil {
+		return fmt.Errorf("Expected secret_id_num_uses %q to be a number, isn't", resp.Data["secret_id_num_uses"])
+	}
+
+	tokenTTL, err := resp.Data["token_ttl"].(json.Number).Int64()
+	if err != nil {
+		return fmt.Errorf("Expected token_ttl %q to be a number, isn't", resp.Data["token_ttl"])
+	}
+
+	tokenNumUses, err := resp.Data["token_num_uses"].(json.Number).Int64()
+	if err != nil {
+		return fmt.Errorf("Expected token_num_uses %q to be a number, isn't", resp.Data["token_num_uses"])
+	}
+
+	tokenMaxTTL, err := resp.Data["token_max_ttl"].(json.Number).Int64()
+	if err != nil {
+		return fmt.Errorf("Expected token_max_ttl %q to be a number, isn't", resp.Data["token_max_ttl"])
+	}
+
+	period, err := resp.Data["period"].(json.Number).Int64()
+	if err != nil {
+		return fmt.Errorf("Expected period %q to be a number, isn't", resp.Data["period"])
+	}
+
+	d.Set("backend", backend)
+	d.Set("role_name", role)
+	d.Set("period", period)
+	d.Set("policies", policies)
+	d.Set("bound_cidr_list", cidrs)
```

Comment:
Nit: Should check err return of `d.Set` for non-scalar values, e.g. sets, lists, maps

Reply:
👍